File: src/main/java/ru/serdtsev/homemoney/HmApplication.java
package ru.serdtsev.homemoney;

import net.sf.ehcache.CacheManager;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import org.springframework.boot.web.servlet.ServletComponentScan;
import org.springframework.context.annotation.Configuration;

@ServletComponentScan
@Configuration
@SpringBootApplication
public class HmApplication {

  @Autowired
  public HmApplication() {
    CacheManager.newInstance();
  }

  public static void main(String[] args) {
    SpringApplication.run(HmApplication.class, args);
  }
}

File: src/main/kotlin/ru/serdtsev/homemoney/HmApplication.kt
package ru.serdtsev.homemoney

import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.web.servlet.ServletComponentScan
import org.springframework.cache.annotation.EnableCaching
import org.springframework.context.annotation.Configuration
import org.springframework.scheduling.annotation.EnableAsync

@ServletComponentScan
@Configuration
@EnableAsync
@EnableCaching
@SpringBootApplication
class HmApplication {
    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            SpringApplication.run(HmApplication::class.java, *args)
        }
    }
}